tasmota/tasmota_template.h

```diff
@@ -716,6 +719,10 @@ const uint8_t kGpioNiceList[] PROGMEM = {
   GPIO_TELEINFO_RX,
   GPIO_TELEINFO_ENABLE,
 #endif
+#ifdef USE_MIEL_HVAC
+  GPIO_MIEL_HVAC_TX,
```

Comment:
I'm afraid we can't afford new GPIOs right now as we are very close to the limit. Unless Theo has a magic way to extent the number of GPIOs, could you reuse existing ones?

Reply:
> I'm afraid we can't afford new GPIOs right now as we are very close to the limit. Unless Theo has a magic way to extent the number of GPIOs, could you reuse existing ones?

If using more GPIOs for another serial protocol is a problem, I think it should be solved more generally by the project. A useful reference point could be line disciplines on unixish systems. That's where a single serial driver always attaches to the hardware, and then the operator attaches a software driver called a line discipline that filters or takes over communication on the serial port.

Applying that here, Tasmota would only have the Serial Tx and Serial Rx GPIOs you could configure on physical pins. Then you'd deprecate others like the TuyaMCU Tx/Rx, ZigBee serial thing(?), and so on, but then add a config option to say what's on the other end of the serial port. The serial port would be like a virtual GPIO pin you attach a protocol to. Or you could rely on the module type to say how the serial port is used.

Reply:
I will leave @arendst respond to that. Most of your code is compiled only if you enable the `#define USE_MIEL_HVAC` so whatever you include will not impact other users. For GPIOs it's different, since the GPIO numbers are reserved even if the user don't enable the feature. We indeed dedicate GPIOs for the most used features like Tuya and Zigbee.